ddns/config/file.py

```diff
@@ -54,15 +55,71 @@ def load_config(config_path):
     except Exception as e:
         stderr.write("Failed to load config file `%s`: %s\n" % (config_path, e))
         raise
-    # flatten the config if it contains nested structures
-    flat_config = {}
-    for k, v in config.items():
-        if isinstance(v, dict):
-            for subk, subv in v.items():
-                flat_config["{}_{}".format(k, subk)] = subv
-        else:
-            flat_config[k] = v
-    return flat_config
+
+    # 处理配置格式：v4.1 providers格式、对象包含configs数组、或单个对象
+    if isinstance(config, dict) and "providers" in config and isinstance(config["providers"], list):
```

Comment:
remove ` isinstance(config, dict) ` check, the config should always be an object